### ISSUE TITLE
deployment issue with transfer learning

### ISSUE THREAD

**Issue by irum** [GPflow/GPflow#1521]
Hi,
I am using transfer learning technique and added my custom layers in it . When I use the prediction from last layer of my transfer learning as input to GPflow model , It trains well but when I use saved checkpoints from the model for deployment then it does not work. I have checked the hyper-parameters after training and during deployment , values are same. So it rules out the possibility that model is not saved properly as checkpoints. 
If I remove my custom layers from base-model of ResNet50v2 then I can successfully deploy the model on test images. Then it gives the same accuracy as I get after training on test images.

Please can you identify the reason behind it by just making a simple example. Otherwise, Please can you send me a confidential link where i can send my code.  I do not understand what technical issue is in the program as i mentioned earlier it works if I just use base-model but does not work on custom layers.

## System information

* GPflow version: 2.0.4
* GPflow installed from: installed by cloning from GitHub and used command `python setup.py install` 
* TensorFlow version: 2.2.0
* Python version  : 3.6
* Operating system: linux

**Comment by irum** [comment]
Hi,
Can anyone please help me on this. I have been stuck on this for long.

**Comment by aidanscannell** [comment]
Hi Irum,

It is hard to fully understand your problem without having example code and/or a stack trace. However, it may be due to how you save the model as Tensorflow does not support saving instances of `gpflow.base.Parameter`. Maybe [check out this notebook](https://gpflow.readthedocs.io/en/master/notebooks/intro_to_gpflow2.html#TensorFlow-saved_model) as it details the problem and a simple workaround. If this is not the issue then I would suggest creating a minimal working example that produces the same error so that it is easy for someone else to understand.

Best,
Aidan

**Comment by irum** [comment]
Hi Aidan,
I am saving the checkpoints and then restoring it as mentioned in  the [notebook](https://gpflow.readthedocs.io/en/master/notebooks/intro_to_gpflow2.html#Checkpointing)
I understand that when I add custom layers during transfer learning (CNN) then my code does not work . If its only in the following form then it works 

            model = resnet50V2()

**Comment by irum** [comment]
It is mistakenly closed. I reopened it

**Comment by st--** [comment]
Hi @irum, GPflow is an open-source project maintained by volunteers - I'm afraid we don't have the capacity to debug other people's code. We try and respond promptly to bug reports and help out to the best of our abilities, but I don't think what you are describing here is something we can fix through changes to GPflow-core. Without a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) that showcases more clearly what the issue is, there's not much others can do to help you.
